crates/ruff_linter/src/rules/flake8_pyi/rules/bad_generator_return_type.rs

```diff
@@ -48,20 +50,40 @@ use crate::checkers::ast::Checker;
 ///     def __iter__(self) -> Iterator[str]:
 ///         yield from "abdefg"
 /// ```
+///
+/// ## Fix safety
+/// This rule tries hard to avoid false-positive errors, and the rule's fix
+/// should always be safe for `.pyi` stub files. However, there is a slightly
+/// higher chance that a false positive might be emitted by this rule when
+/// applied to runtime Python (`.py` files). As such, the fix is marked as
+/// unsafe for any `__iter__` or `__aiter__` method in a `.py` file that has
+/// more than two statements (including docstrings) in its body.
 #[violation]
 pub struct GeneratorReturnFromIterMethod {
-    better_return_type: String,
-    method_name: String,
+    return_type: Iterator,
+    method: Method,
```

Comment:
@AlexWaygood - I changed this to use the enums, which implement `std::fmt::Display`, so we can still inline them in the `format!` call and automatically get the expected formatting.